backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/nodes/AbstractUsageNodeDescriptionProvider.java

```diff
@@ -76,6 +79,18 @@ protected String getNodeDescriptionName() {
         return this.getDescriptionNameGenerator().getNodeName(this.eClass);
     }
 
+    /**
+     * Set an optional precondition for the future node.
+     *
+     * @param aPrecondition
+     *            a precondition (or <code>null</code> to unset the precondition).
+     * @return this for convenience
+     */
+    public AbstractUsageNodeDescriptionProvider setPrecondition(String aPrecondition) {
```

Comment:
It would be better to have a dedicated constructor allowing to set the precondition.
You also must initialize `precondition` to `null` in the default constructor.

Reply:
Discussed with you. I think this methods is more flexible.